ee/tables/security/cert_trust_settings.go

```diff
@@ -0,0 +1,361 @@
+	}
+	expectedTrustSettingsCountStr := matches[0][1]
+	expectedTrustSettingsCount, err := strconv.Atoi(expectedTrustSettingsCountStr)
+	if err != nil {
+		return nil, fmt.Errorf("parsing number of trust settings: %w", err)
+	}
+
+	// Parse the expected number of trust settings from our reader.
+	trustSettings := make([]*trustSetting, expectedTrustSettingsCount)
+	for i := range expectedTrustSettingsCount {
+		trustSetting, err := parseTrustSetting(reader)
+		if err != nil {
+			return nil, fmt.Errorf("parsing trust setting %d of %d: %w", i, expectedTrustSettingsCount, err)
+		}
+		trustSettings[i] = trustSetting
+	}
+
+	return trustSettings, nil
+}
+
+// parseTrustSetting parses the upcoming trust setting from the given reader.
+// The trust setting, in context, will look like this:
+//
+//	Cert 0: Example Root CA
+//	   Number of trust settings : 1
+//	   Trust Setting 0:
+//	      Policy OID            : SSL
+//	      Allowed Error         : CSSMERR_TP_CERT_EXPIRED
+//	      Result Type           : kSecTrustSettingsResultTrustRoot
+//	Cert 1: ...
+//
+// Here, we attempt to parse Policy OID, Allowed Error, and Result Type. The function terminates
+// when it peeks ahead and sees either another trust setting (Trust Setting 1) or another cert (Cert 1: ...).
+func parseTrustSetting(reader *bufio.Reader) (*trustSetting, error) {
+	// Read the first line, which will be a header matching `trustSettingHeaderRegexp`
+	// indicating our current index (e.g. "Trust Setting 0".)
+	currentLine, err := reader.ReadString(lineDelimiter)
+	if err != nil {
+		return nil, fmt.Errorf("reading first line of trust setting: %w", err)
+	}
+	currentLine = strings.TrimSpace(currentLine)
+
+	// We expect the first line to match `trustSettingHeaderRegexp``. We don't actually need
+	// to extract the index here, but we want to make sure that we haven't gotten off with parsing
+	// anywhere.
+	matches := trustSettingHeaderRegexp.FindAllStringSubmatch(currentLine, -1)
+	if len(matches) < 1 || len(matches[0]) < 2 {
+		return nil, fmt.Errorf("no trust setting: %s", currentLine)
+	}
+
+	// Read trust setting info, peeking ahead first to see whether we've completed reading this setting.
+	t := trustSetting{}
+	for {
+		// We peek ahead here to see whether we've already finished reading this setting.
+		// We check for a line starting with `certHeaderPrefix` (indicating this was the last
+		// trust setting in our list of trust settings, and we've advanced to the next cert), or
+		// `trustSettingHeaderPrefix` (indicating this was not the last trust setting in our list, and
+		// we've advanced to the next trust setting). The longer of these strings is `trustSettingHeaderPrefix`,
+		// typically with 3 spaces before it. So, we peek ahead len(trustSettingHeaderPrefix)+10, to be safe.
+		nextBytes, err := reader.Peek(len(trustSettingHeaderPrefix) + 10)
+		if err != nil && !errors.Is(err, bufio.ErrBufferFull) && !errors.Is(err, io.EOF) {
```

Comment:
if there is an `io.EOF` err here should we break before attempting to parse nextBytes?

Reply:
No, because we might have one more line that's under 30 chars left to read -- I will leave a comment